azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
+        Executors.newFixedThreadPool(activeExecutors.size());
```

Comment:
We should never have a activeExecutors.size() =  0 as we fail AzkabanWebServer, if there is no activeExecutors and we do not update activeExecutors if new executor set is empty.